### PR TITLE
homework 8

### PR DESCRIPTION
**Pull request by artsiom-tsaryonau**

### CODE REVIEW

File: homework8/build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+dependencies {
+    compile('org.springframework.boot:spring-boot-starter-data-jpa')
+    compile('org.springframework.boot:spring-boot-starter-data-rest')
+    compile('org.springframework.boot:spring-boot-starter-web')
+    compile('it.ozimov:spring-boot-email-core:0.6.2')
+    runtime('com.h2database:h2')
+    providedRuntime('org.springframework.boot:spring-boot-starter-tomcat')
+    testCompile('org.springframework.boot:spring-boot-starter-test')
+}
```
Comment (AShynkevich) [inline homework8/build.gradle]: Add new line

File: homework8/src/main/java/com/epam/lab/mentoring/BookViewApplication.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@ComponentScan(basePackages = { "com.epam.lab.mentoring" })
+@EnableEmailTools
+@EnableAsync
+public class BookViewApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(BookViewApplication.class, args);
+    }
+}
+
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/BookViewApplication.java]: It seems to me here is an extra line

File: homework8/src/main/java/com/epam/lab/mentoring/ServletInitializer.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+public class ServletInitializer extends SpringBootServletInitializer {
+
+    @Override
+    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+        return application.sources(BookViewApplication.class);
+    }
+
+}
+
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/ServletInitializer.java]: Extra line

File: homework8/src/main/java/com/epam/lab/mentoring/repository/filesystem/FileSystemUtils.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            }
+        } catch (IOException exc) {
+            log.error("Failed to create new file [{}] in [{}].", filename, directory, exc);
+        }
+    }
+
+    private static String makePath(String folder, String filename) {
+        return folder + File.separator + filename;
+    }
+
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/repository/filesystem/FileSystemUtils.java]: Remove the line

File: homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.epam.lab.mentoring.repository.filesystem.IBooksFileRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.io.IOException;
+
+@RestController
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: It looks weird. I don't think that is properly controller annotation.

File: homework8/src/main/java/com/epam/lab/mentoring/repository/web/IBooksRepository.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.epam.lab.mentoring.repository.web;
+
+import com.epam.lab.mentoring.domain.Book;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/repository/web/IBooksRepository.java]: Do you really need this Rest annotation?

File: homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            if (!file.isEmpty()) {
+                fileRepository.writeFile(file.getOriginalFilename(), file.getBytes());
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return new ModelAndView("redirect:/books-home");
+    }
+
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: Remove the line

File: homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    @PostMapping("/upload")
+    public ModelAndView uploadFile(@RequestParam("file") MultipartFile file) {
+        try {
+            if (!file.isEmpty()) {
+                fileRepository.writeFile(file.getOriginalFilename(), file.getBytes());
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
```
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: Remove the line

File: homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java (hunk 7 above)
Comment (AShynkevich) [inline homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: Extra line

File: homework8/build.gradle (hunk 1 above)
Comment (artsiom-tsaryonau) [reply homework8/build.gradle]: fixed

File: homework8/src/main/java/com/epam/lab/mentoring/BookViewApplication.java (hunk 2 above)
Comment (artsiom-tsaryonau) [reply homework8/src/main/java/com/epam/lab/mentoring/BookViewApplication.java]: fixed

File: homework8/src/main/java/com/epam/lab/mentoring/ServletInitializer.java (hunk 3 above)
Comment (artsiom-tsaryonau) [reply homework8/src/main/java/com/epam/lab/mentoring/ServletInitializer.java]: fixed

File: homework8/src/main/java/com/epam/lab/mentoring/repository/filesystem/FileSystemUtils.java (hunk 4 above)
Comment (artsiom-tsaryonau) [reply homework8/src/main/java/com/epam/lab/mentoring/repository/filesystem/FileSystemUtils.java]: fixed

File: homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java (hunk 8 above)
Comment (artsiom-tsaryonau) [reply homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: fixed

File: homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java (hunk 7 above)
Comment (artsiom-tsaryonau) [reply homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: fixed

Comment (artsiom-tsaryonau) [reply homework8/src/main/java/com/epam/lab/mentoring/rest/BooksWebController.java]: fixed

**Pull request merged**